### PR TITLE
suppress warning for unannotated routes

### PR DESCRIPTION
**Pull request by seaingsky**

### CODE REVIEW

File: app/controllers/ApiDocController.scala [hunk 1]
```diff
@@ -152,36 +152,42 @@ class ApiDocController extends Controller {
...
+  def validate(routeEntries: List[RouteEntry]): List[RouteEntry] = {
     val alreadyIncluded = scala.collection.mutable.Set[String]()
+    val list = new java.util.ArrayList[RouteEntry]()
 
     routeEntries.foreach(routeEntry => {
 
-      if (!hasMethodAnnotation(routeEntry.scalaClass, routeEntry.scalaMethod))
-        throw new Exception(s"Missing ApiDoc for ${routeEntry.scalaClass}.${routeEntry.scalaMethod} (Make sure the Class is annotated, and not the corresponding Object)")
+      if (hasMethodAnnotation(routeEntry.scalaClass, routeEntry.scalaMethod)) {
+        list.add(routeEntry)
```
Comment (dspeele) [inline app/controllers/ApiDocController.scala]: you could do something a bit cleaner here like: 

for (routeEntry <- routeEntries; if (hasMethodAnnotation(routeEntry.scalaClass, routeEntry.scalaMethod))) yield routeEntry

Comment (anoopriya) [inline app/controllers/ApiDocController.scala]: Taking a look at the code with Hugh... Please do not merge

File: app/controllers/ApiDocController.scala [hunk 2]
```diff
@@ -152,36 +152,42 @@ class ApiDocController extends Controller {
     expandIncludes(annotation.doc, alreadyIncluded)
   }
 
-  def validate(routeEntries: List[RouteEntry]): Unit = {
+  def validate(routeEntries: List[RouteEntry]): List[RouteEntry] = {
     val alreadyIncluded = scala.collection.mutable.Set[String]()
+    val list = new java.util.ArrayList[RouteEntry]()
```
Comment (anoopriya) [inline app/controllers/ApiDocController.scala]: You can get rid of this declaration. You can use either for-comprehension(as Dana suggested) or filter() (Hugh's recommendation) which will yield back a list of RouteEntry

File: app/controllers/ApiDocController.scala [hunk 3]
```diff
@@ -152,36 +152,42 @@ class ApiDocController extends Controller {
...
-        throw new UriMismatchException(s"Conflicting uri declared in the autodoc and in conf/routes for ${routeEntry.scalaClass}.${routeEntry.scalaMethod}.\n"+
-                                       s"autodoc: $jsonUri. conf/routes: ${routeEntry.uri}")
+        if (!hasSameUri(jsonUri, routeEntry.uri))
+          throw new UriMismatchException(s"Conflicting uri declared in the autodoc and in conf/routes for ${routeEntry.scalaClass}.${routeEntry.scalaMethod}.\n" +
+            s"autodoc: $jsonUri. conf/routes: ${routeEntry.uri}")
+
+      }
     })
+
+    list.toArray.toList.asInstanceOf[List[RouteEntry]]
```
Comment (anoopriya) [inline app/controllers/ApiDocController.scala]: Similar to above, you can get rid of this unnecessary java to scala conversion. Use for or filter() which will yield the list

File: app/controllers/ApiDocController.scala (hunk 1 above)
Comment (diamonddave) [inline app/controllers/ApiDocController.scala]: Why yes, we can spend an infinite amount of time optimizing this in various ways, however we many miles to go before we sleep. There are no infinite loops, no core dumps, no memory leaks, no concurrency problems. We can take any suggestions as a backlog when we have EVERY feature  realized.

**Pull request merged**